File: internal/flow/run_test.go
package flow

import (
	"math/rand"
	"strings"
	"testing"
)

func TestRandomSpriteNameWithPrefix(t *testing.T) {
	r := rand.New(rand.NewSource(1))
	name := randomSpriteName("repo-branch", r)
	if !strings.HasPrefix(name, "repo-branch-") {
		t.Fatalf("expected prefix, got %q", name)
	}
	if strings.Count(name, "-") < 3 {
		t.Fatalf("expected adjective/noun suffix, got %q", name)
	}
}

func TestRandomSpriteNameNoPrefix(t *testing.T) {
	r := rand.New(rand.NewSource(2))
	name := randomSpriteName("", r)
	if strings.HasPrefix(name, "-") || strings.HasSuffix(name, "-") {
		t.Fatalf("invalid name formatting: %q", name)
	}
	if strings.Count(name, "-") != 1 {
		t.Fatalf("expected adjective-noun form, got %q", name)
	}
}

func TestSanitizeRepoURLHTTPRemovesUserinfo(t *testing.T) {
	g := sanitizeRepoURL("https://user:secret@example.com/repo.git")
	if g != "https://example.com/repo.git" {
		t.Fatalf("unexpected sanitized url: %s", g)
	}
}

func TestSanitizeRepoURLScpStyleRemovesUser(t *testing.T) {
	g := sanitizeRepoURL("git@github.com:owner/repo.git")
	if g != "github.com:owner/repo.git" {
		t.Fatalf("unexpected sanitized url: %s", g)
	}
}

File: test/flow/run.test.ts
import { describe, it } from "node:test";
import * as assert from "node:assert/strict";
import { randomSpriteName, sanitizeRepoURL } from "../../src/flow/run.js";

describe("randomSpriteName", () => {
  it("includes prefix", () => {
    const name = randomSpriteName("repo-branch");
    assert.ok(name.startsWith("repo-branch-"), `expected prefix, got ${name}`);
    assert.ok(name.split("-").length >= 4, `expected adjective/noun suffix, got ${name}`);
  });

  it("works without prefix", () => {
    const name = randomSpriteName("");
    assert.ok(!name.startsWith("-"), `invalid name formatting: ${name}`);
    assert.ok(!name.endsWith("-"), `invalid name formatting: ${name}`);
    assert.equal(name.split("-").length, 2, `expected adjective-noun form, got ${name}`);
  });
});

describe("sanitizeRepoURL", () => {
  it("removes userinfo from HTTP URL", () => {
    const g = sanitizeRepoURL("https://user:secret@example.com/repo.git");
    assert.equal(g, "https://example.com/repo.git");
  });

  it("removes user from scp-style URL", () => {
    const g = sanitizeRepoURL("git@github.com:owner/repo.git");
    assert.equal(g, "github.com:owner/repo.git");
  });
});
